internal/controller/garbagecollector_controller.go

```diff
@@ -273,10 +274,13 @@ func (r *GarbageCollectorReconciler) Reconcile(ctx context.Context, _ ctrl.Reque
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *GarbageCollectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if r.Name == "" {
+		r.Name = "nonadmingarbagecollector"
+	}
```

Comment:
This is only needed if a controller has no `For(&nacv1alpha1.TYPE{})` where name can be gathered from GroupVersionKind ie. prior default behavior... ie. where you had to Named("<name>").